### PR TITLE
Improve readmes

### PR DESCRIPTION
**Pull request by nventuro**
We've received multiple reports of people having trouble navigating the repository due to poor readmes, and particularly having trouble understanding the deployments package. This PR aims to improve that, as well as document our deployment process somewhat.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,23 +19,40 @@ Active development occurs in this repository, which means some contracts in it m
...
+- [`v2-interfaces`](./pkg/interfaces): Solidity interfaces for all contracts.
 - [`v2-vault`](./pkg/vault): the [`Vault`](./pkg/vault/contracts/Vault.sol) contract and all core interfaces, including [`IVault`](./pkg/vault/contracts/interfaces/IVault.sol) and the Pool interfaces: [`IBasePool`](./pkg/vault/contracts/interfaces/IBasePool.sol), [`IGeneralPool`](./pkg/vault/contracts/interfaces/IGeneralPool.sol) and [`IMinimalSwapInfoPool`](./pkg/vault/contracts/interfaces/IMinimalSwapInfoPool.sol).
-- [`v2-pool-weighted`](./pkg/pool-weighted): the [`WeightedPool`](./pkg/pool-weighted/contracts/WeightedPool.sol) and [`WeightedPool2Tokens`](./pkg/pool-weighted/contracts/WeightedPool2Tokens.sol) contracts, along with their associated factories.
+- [`v2-pool-weighted`](./pkg/pool-weighted): the [`WeightedPool`](./pkg/pool-weighted/contracts/WeightedPool.sol), [`WeightedPool2Tokens`](./pkg/pool-weighted/contracts/WeightedPool2Tokens.sol) and [`LiquidityBootstrappingPool`](./pkg/pool-weighted/contracts/smart/LiquidityBootstrappingPool.sol) ontracts, along with their associated factories.
```
Comment (EndymionJkb) [inline README.md:24]:
```suggestion
- [`v2-pool-weighted`](./pkg/pool-weighted): the [`WeightedPool`](./pkg/pool-weighted/contracts/WeightedPool.sol), [`WeightedPool2Tokens`](./pkg/pool-weighted/contracts/WeightedPool2Tokens.sol) and [`LiquidityBootstrappingPool`](./pkg/pool-weighted/contracts/smart/LiquidityBootstrappingPool.sol) contracts, along with their associated factories.
```

File: pkg/deployments/DEPLOYING.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Deploying Balancer Contracts
+
+The deployments package aims to provide a robust mechanism to run and verify deployments on multiple networks. This document describes rationale, what guarantees it provides, and how to create and run deployments.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:3]:
```suggestion
The deployments package aims to provide a robust mechanism to run and verify deployments on multiple networks. This document describes the rationale, what guarantees it provides, and how to create and run deployments.
```

File: pkg/deployments/DEPLOYING.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+## Overview
+
+Each deployment unit is called a 'task'. Tasks have a unique ID, which is often prefixed with the date in which they were first executed (to e.g. tell apart the deployment of Weighted Pools from April 2021, and the one in July 2022, which might feature widly different behavior).
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:7]:
```suggestion
Each deployment unit is called a 'task'. Tasks have a unique ID, which is often prefixed with the date on which they were first executed (to e.g. differentiate the April 2021 deployment of Weighted Pools from the one deployed in July 2022, which might have wildly different behavior).
```

File: pkg/deployments/DEPLOYING.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+A task is made up of multiple components:
+
+- one or multiplle build information JSON files, located in the `build-info` directory.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:11]:
```suggestion
- one or multiple build information JSON files, located in the `build-info` directory.
```

File: pkg/deployments/DEPLOYING.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- an `input.ts` file, containing information on dependencies and deployment parameters (which might differ accross networks).
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:12]:
```suggestion
- an `input.ts` file, containing information on dependencies and deployment parameters (which might differ across networks).
```

File: pkg/deployments/DEPLOYING.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+When Hardhat compiles a Solidity project, in addition to the traditional contract artifacts it also produces a special file in the `artifacts/build-info` directory, which includes all of the compilation inputs (full source code, compiler version and settings, etc.) and all of the compilation outputs (creation bytecode, ABI, AST, etc.). By storing this file in the repository and using it as the source of truth for deployments, we fully break the dependency on the source code and are able to modify it and update it after a deployment is created, while guaranteeing that all deployments of a given task will be identical (since they share the same build information).
+
+As part of our development process, we frequently alter the source code of contracts that have already been deployed, and keep no historical versions: the deployment data is fully stored in the build information file.
+
+## Creating Tasks
+
+A new task is created by placing a new directory named after the task ID in the `tasks`, and populating the `build-info` directory and writing appropiate `input.ts` and `index.ts` files. The `output`, `abi` and `bytecode` directories are automatically generated and populated.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:25]:
```suggestion
A new task is created by placing a new directory named after the task ID in the `tasks` directory, and populating the `build-info` directory and writing appropriate `input.ts` and `index.ts` files. The `output`, `abi` and `bytecode` directories are automatically generated and populated by running the extract-artifacts script after the build-info is present.
```

File: pkg/deployments/DEPLOYING.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+The build information should be generated on a clean commit, so that it can be verified by other parties. On the project directory, delete the `artifacts` directory and run `yarn hardhat compile` for a clean compilation. The file will be then be located at `artifacts/build-info`.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:27]: Do we need to explain about editing package.json to remove the part that normally deletes build-info on compilation? Or make a compile-with-buildinfo?

File: pkg/deployments/DEPLOYING.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+Inputs and task instructions are plain TypeScript files that call appropriate functions - there's no DSL. The recommended way to write them is to copy the structure from the `input.ts` and `index.ts` files from a similar task, and then edit those as needed.
+
+### Vyper Compilation
+
+Hardhat does not generate build information for Vyper contracts - those must be created manually.
+
+## Verifying Tasks
+
+There's three things to verify from a task:
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:37]:
```suggestion
There are three things to verify from a task:
```

File: pkg/deployments/DEPLOYING.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+- the build information
+- the inputs
+- the deployment script
+
+The last two can be performed with simple manual inspection, that is, reviewing `input.ts` and `index.ts` and checking that inputs are sensible and constructor arguments provided correctly.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:43]:
```suggestion
The last two can be performed with simple manual inspection, that is, reviewing `input.ts` and `index.ts` and checking that they have sensible inputs and correct constructor arguments.
```

File: pkg/deployments/DEPLOYING.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+## Checking Deployments
+
+The deployments package is able to verify that the addresses found in the `output` directories correspond exactly to a run of the task as described by `index.ts`, using the inputs at `input.ts` and the data from the `build-info` directory.
+
+This can be done manually by running the `check-deployments` Hardhat task (`yarn hardhat check-deployments --network <network>`), but it is also done automatically on CI. Like deployment, [this also requires access to an RPC endpoint](#prerequisites).
+
+## Running Tests
+
+Some tasks define fork tests, which consist of creating a local form from real network and some block, running the task on said fork, and then performing tests on top.
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:121]:
```suggestion
Some tasks define fork tests, which consist of creating a local fork from a real network at a given block, running the task on said fork, and then performing tests.
```

File: pkg/deployments/DEPLOYING.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+Unlike regular deployments, this requires access to an _archive node_ instead of a regular RPC endpoint. [Alchemy](https://www.alchemy.com/) provides access to these on their free tier. They are configured the same way as [regular deployment endpoints](#prerequisites) (it is possible to simply use the archive endpoint for all purposes, including contract deployment).
+
+Once that is setup, fork tests can be run:
```
Comment (EndymionJkb) [inline pkg/deployments/DEPLOYING.md:125]:
```suggestion
Once that is set up, fork tests can be run:
```

File: pkg/deployments/README.md [hunk 12]
```diff
@@ -5,14 +5,18 @@
...
 
-This package contains the addresses and ABIs of all Balancer V2 deployed contracts, for Ethereum, Polygon and Arbitrum mainnet, as well as various test networks. Each deployment consists of a deployment script (called 'task'), inputs (script configuration, such as dependencies), outputs (typically contract addresses), and ABIs of related contracts.
+This package contains the addresses and ABIs of all Balancer V2 deployed contracts, for Ethereum, Polygon, Arbitrum and Optimism mainnet, as well as various test networks. Each deployment consists of a deployment script (called 'task'), inputs (script configuration, such as dependencies), outputs (typically contract addresses), and ABIs of related contracts.
 
-Addresses and ABIs can be used consumed from the package in JavaScript environments, or manually retrieved from the [GitHub](https://github.com/balancer-labs/balancer-v2-monorepo/tree/deployments-latest/pkg/deployments) repository.
+Addresses and ABIs can be used consumed from the package in JavaScript environments, or manually retrieved from the [GitHub](https://github.com/balancer-labs/balancer-v2-monorepo/tree/master/pkg/deployments) repository.
```
Comment (EndymionJkb) [inline pkg/deployments/README.md:10]:
```suggestion
Addresses and ABIs can be consumed from the package in JavaScript environments, or manually retrieved from the [GitHub](https://github.com/balancer-labs/balancer-v2-monorepo/tree/master/pkg/deployments) repository.
```

**Comment by nventuro** [comment]
oh my god i suck at writing

File: pkg/deployments/DEPLOYING.md (hunk 7 above)
Comment (nventuro) [reply pkg/deployments/DEPLOYING.md:27]: `yarn hardhat compile` will do the right thing. I don't want to go into too much detail tbh - this is not necessarily a step-by-step guide. And the reason why we delete the buldinfos is entirely accidental.

**Pull request merged**